### PR TITLE
aon-4.2: add authorization for benefit sponsors employer datatable

### PR DESCRIPTION
**Pull request by vkghub**
https://www.pivotaltracker.com/story/show/185587131

### CODE REVIEW

File: components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb [hunk 1]
```diff
@@ -282,6 +282,11 @@ def nested_filter_definition
         end
         filters
       end
+
+      def authorized?(current_user, _controller, _action, _resource)
+        return true if current_user&.has_hbx_staff_role?
```
Comment (saikumar9) [inline components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb:287]:
```suggestion
        current_user&.has_hbx_staff_role?
```

File: components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb [hunk 2]
```diff
@@ -282,6 +282,11 @@ def nested_filter_definition
...
+        false
```
Comment (saikumar9) [inline components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb:288]: you can remove this.

File: components/benefit_sponsors/spec/models/effective/datatables/benefit_sponsors_employer_datatable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Effective::Datatables::BenefitSponsorsEmployerDatatable, dbclean: :after_each do
+
+  describe '#authorized?' do
```
Comment (saikumar9) [inline components/benefit_sponsors/spec/models/effective/datatables/benefit_sponsors_employer_datatable_spec.rb:7]: Good test coverage.

**Review by saikumar9 — DISMISSED** [review]

File: components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb (hunk 1 above)
Comment (vkghub) [reply components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb:287]: current_user&.has_hbx_staff_role? will return true/false/nil. I made that more specific to true/false .

File: components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb [hunk 4]
```diff
@@ -282,6 +282,10 @@ def nested_filter_definition
         end
         filters
       end
+
+      def authorized?(current_user, _controller, _action, _resource)
+        (current_user.present? && current_user.has_hbx_staff_role?) ? true : false
+      end
```
Comment (TreyE) [inline components/benefit_sponsors/app/models/effective/datatables/benefit_sponsors_employer_datatable.rb:288]: No need for the extra operator,
`current_user.present? && current_user.has_hbx_staff_role?`
is fine on it's own.

**Review by TreyE — CHANGES_REQUESTED** [review]

**Review by TreyE — APPROVED** [review]

**Review by saipraveen18 — APPROVED** [review]

**Pull request merged**